overlord/snapstate/snapmgr.go

```diff
@@ -560,6 +666,13 @@ func (m *SnapManager) Ensure() error {
 
 	m.runner.Ensure()
 
+	// run after the runner so if we get a new core snap, we immediately
+	// generate the matching apparmor profile for snap-confine from core
```

Comment:
indeed I don't understand this comment, also I don't see any kind of locks ? it feels also strange to do all this in snapmgr and not ifacemgr